src/journal-remote/header-util.c

```diff
@@ -0,0 +1,170 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "header-util.h"
+#include "strv.h"
+
+/* HTTP header name can contains:
+- Alphanumeric characters: a-z, A-Z, and 0-9
+- The following special characters: - and _ */
+#define VALID_HEADER_NAME_CHARS \
+        DIGITS LETTERS "_"      \
+                       "-"
+
+#define VALID_HEADER_NAME_LENGTH 40
+
+#define VALID_HEADER_VALUE_CHARS \
+        DIGITS LETTERS "_"       \
+                       " "       \
+                       ":"       \
+                       ";"       \
+                       "."       \
+                       ","       \
+                       "\\"      \
+                       "/"       \
+                       "'"       \
+                       "\""      \
+                       "?"       \
+                       "!"       \
+                       "("       \
+                       ")"       \
+                       "{"       \
+                       "}"       \
+                       "["       \
+                       "]"       \
+                       "@"       \
+                       "<"       \
+                       ">"       \
+                       "="       \
+                       "-"       \
+                       "+"       \
+                       "*"       \
+                       "#"       \
+                       "$"       \
+                       "&"       \
+                       "`"       \
+                       "|"       \
+                       "~"       \
+                       "^"       \
+                       "%"
+
+static bool header_name_is_valid(const char *e, size_t n) {
+        if (!e)
+                return false;
+
+        if (n > VALID_HEADER_NAME_LENGTH)
+                return false;
+
+        for (const char *p = e; p < e + n; p++)
+                if (!strchr(VALID_HEADER_NAME_CHARS, *p))
+                        return false;
+
+        return true;
+}
+
+static bool header_value_is_valid(const char *e) {
+        if (!e)
+                return false;
+
+        int n = strlen_ptr(e);
```

Comment:
simple strlen_ptr() would suffice too

Reply:
you drop 32bit on the floor here. strlen() returns size_t (i.e. typically 64bit), not `int` (typically 32bit). change type to `size_t`